### PR TITLE
Add end to end tests to the 'userConvertCmdF' mmctl command

### PR DESCRIPTION
**Pull request by gupsho**
#### Summary
Add end to end tests to the 'userConvertCmdF' mmctl command

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15669

### CODE REVIEW

File: commands/user_e2e_test.go [hunk 1]
```diff
@@ -136,3 +136,46 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+
+		email := s.th.BasicUser.Email
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("bot", true, "")
+
+		err := userConvertCmdF(s.th.SystemAdminClient, cmd, []string{email})
+		s.Require().NoError(err)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:165]:
```suggestion
	s.RunForSystemAdminAndLocal("Valid user to bot convert", func() {
		printer.Clean()

		email := s.th.BasicUser.Email
		cmd := &cobra.Command{}
		cmd.Flags().Bool("bot", true, "")

		err := userConvertCmdF(c, cmd, []string{email})
		s.Require().NoError(err)
		s.Require().Len(printer.GetLines(), 1)
		s.Require().Len(printer.GetErrorLines(), 0)
	})
```

File: commands/user_e2e_test.go [hunk 2]
```diff
@@ -136,3 +136,46 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForAllClients("Error when no flag provided", func(c client.Client) {
+		printer.Clean()
+
+		emailArg := "example@example.com"
+		cmd := &cobra.Command{}
+
+		err := userConvertCmdF(c, cmd, []string{emailArg})
+		s.Require().Error(err)
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:150]: Shall we check if we have the expected error here?

File: commands/user_e2e_test.go [hunk 3]
```diff
@@ -136,3 +136,46 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
...
+	s.Run("Valid user to bot convert", func() {
+		printer.Clean()
+
+		email := s.th.BasicUser.Email
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("bot", true, "")
+
+		err := userConvertCmdF(s.th.SystemAdminClient, cmd, []string{email})
+		s.Require().NoError(err)
+		s.Require().Len(printer.GetLines(), 1)
```
Comment (isacikgoz) [inline commands/user_e2e_test.go:163]: Shall we check if we have the expected output here?

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Hey @gupsho, thanks for the contribution!! 🎉 I have some change requests to improve the tests. Also, what do you think about adding such cases that satisfies some errors where the user/bot can't be found etc? Or, regular user tries to convert user/bot?

File: commands/user_e2e_test.go (hunk 2 above)
Comment (gupsho) [reply commands/user_e2e_test.go:150]: Done.

File: commands/user_e2e_test.go (hunk 3 above)
Comment (gupsho) [reply commands/user_e2e_test.go:175]: Done.

File: commands/user_e2e_test.go (hunk 1 above)
Comment (gupsho) [reply commands/user_e2e_test.go]: I am not able to use same user with both clients. So created a separate test for LocalClient and SystemAdminClient.

**Comment by gupsho** [comment]
> Hey @gupsho, thanks for the contribution!! 🎉 I have some change requests to improve the tests. Also, what do you think about adding such cases that satisfies some errors where the user/bot can't be found etc? Or, regular user tries to convert user/bot?



> Hey @gupsho, thanks for the contribution!! 🎉 I have some change requests to improve the tests. Also, what do you think about adding such cases that satisfies some errors where the user/bot can't be found etc? Or, regular user tries to convert user/bot?

Done.

File: commands/user_e2e_test.go (hunk 1 above)
Comment (gupsho) [reply commands/user_e2e_test.go]: Done.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Comment by gupsho** [comment]
@isacikgoz Please let me know if you want me to make any other changes.

**Review by isacikgoz — APPROVED** [review]
Thanks @gupsho this is awesome! Sorry for the late review, somehow this went out of my eye.

**Comment by hanzei** [comment]
Sorry @lindy65, I merged this PR by mistake. Do I need to revert the merge or is it good to go=

**Comment by lindy65** [comment]
No problem @hanzei - these are tested after merging anyway so the QA check on `mmctl` PRs is required to make notes on the associated Jira ticket.  I'll do that now 👍

**Pull request merged**